pkg/kubectl/cmd/testing/fake.go

```diff
@@ -680,6 +685,7 @@ func testDynamicResources() []*discovery.APIGroupResources {
 			VersionedResources: map[string][]metav1.APIResource{
 				"v1beta1": {
 					{Name: "deployments", Namespaced: true, Kind: "Deployment"},
+					{Name: "thirdpartyresources", Namespaced: false, Kind: "ThirdPartyResource"},
```

Comment:
If a third party resource was registered, we would expect it to show up in discovery. Add another API group stanza for "mygroup.com" with a single version and a single namespaced resource "foos" of kind Foo